src/Conf.ml

```diff
@@ -893,9 +897,20 @@ let kind_of fname =
   | ".mlt" -> `Use_file
   | _ -> !kind
 
+let update_using_xdg =
+  match Caml.Sys.getenv_opt "XDG_CONFIG_HOME" with
+  | Some xdg_config_home ->
+      let filename =
+        Filename.concat xdg_config_home "ocamlformat/.ocamlformat"
+      in
+      Staged.stage (fun conf -> read_config ~filename ~parents:false conf)
+  | None -> Staged.stage (fun conf -> conf)
```

Comment:
@gpetiot how about like this?

Reply:
Better of course. Did not know `Staged`, guess I did not use it the best way :)